org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/validation/AddMarkersOperation.java

```diff
@@ -38,34 +52,50 @@
 	private final ImmutableList<String> markerIds;
 
 	private MarkerCreator markerCreator;
-
 	private final MarkerTypeProvider markerTypeProvider;
+	private IAnnotationModel annotationModel;
 
 	/**
-	 * @deprecated use {@link AddMarkersOperation#AddMarkersOperation(IResource, List, Set, boolean, MarkerCreator, MarkerTypeProvider)} instead
+	 * @deprecated use
+	 *             {@link AddMarkersOperation#AddMarkersOperation(IResource, List, Set, boolean, IAnnotationModel, MarkerCreator, MarkerTypeProvider)}
+	 *             instead
 	 */
 	@Deprecated
-	public AddMarkersOperation(IResource resource, List<Issue> issues, String markerId,
-			boolean deleteMarkers, MarkerCreator markerCreator) {
+	public AddMarkersOperation(IResource resource, List<Issue> issues, String markerId, boolean deleteMarkers,
+			MarkerCreator markerCreator) {
 		this(resource, issues, Collections.singleton(markerId), deleteMarkers, markerCreator);
 	}
-	
+
 	/**
-	 * @deprecated use {@link AddMarkersOperation#AddMarkersOperation(IResource, List, Set, boolean, MarkerCreator, MarkerTypeProvider)} instead
+	 * @deprecated use
+	 *             {@link AddMarkersOperation#AddMarkersOperation(IResource, List, Set, boolean, IAnnotationModel, MarkerCreator, MarkerTypeProvider)}
+	 *             instead
 	 */
 	@Deprecated
-	public AddMarkersOperation(IResource resource, List<Issue> issues, Set<String> markerIds,
-			boolean deleteMarkers, MarkerCreator markerCreator) {
+	public AddMarkersOperation(IResource resource, List<Issue> issues, Set<String> markerIds, boolean deleteMarkers,
+			MarkerCreator markerCreator) {
 		this(resource, issues, markerIds, deleteMarkers, markerCreator, new MarkerTypeProvider());
 	}
-	
+
 	/**
-	 * @since 2.3
+	 * @deprecated use
+	 *             {@link AddMarkersOperation#AddMarkersOperation(IResource, List, Set, boolean, IAnnotationModel, MarkerCreator, MarkerTypeProvider)}
+	 *             instead
 	 */
-	public AddMarkersOperation(IResource resource, List<Issue> issues, Set<String> markerIds,
-			boolean deleteMarkers, MarkerCreator markerCreator, MarkerTypeProvider markerTypeProvider) {
+	@Deprecated
+	public AddMarkersOperation(IResource resource, List<Issue> issues, Set<String> markerIds, boolean deleteMarkers,
+			MarkerCreator markerCreator, MarkerTypeProvider markerTypeProvider) {
+		this(resource, issues, markerIds, deleteMarkers, null, markerCreator, markerTypeProvider);
+	}
+
+	/**
+	 * @since 2.14
+	 */
+	public AddMarkersOperation(IResource resource, List<Issue> issues, Set<String> markerIds, boolean deleteMarkers,
```

Comment:
Time to introduce a parameter object?

Reply:
I just experimented a bit. Of course it is possible to introduce some parameter object or a builder. But because of the backwards compatibility (the old constructors) the telescoping constructor remains and in result the code gets just more complicated. If it is ok with you I would prefer to leave it as it is.